doc/specs/#140 - ZIP Support.md

```diff
@@ -0,0 +1,135 @@
+---
+author: Alexis Bekhdadi @midoriiro
+created on: 2021-11-16
+last updated: 2021-11-16
+issue id: 140
+---
+
+# Spec Title
+
+For [#140](https://github.com/microsoft/winget-cli/issues/140)
+
+## Abstract
+
+This feature adds to winget-cli the ability to install application delivered through a ZIP archive.
+
+## Inspiration
+
+A bunch of applications are only provided by developer's through a ZIP archive. There are three cases:
+1) The archive contain only one binary that need do to be extracted and add to PATH env var
+2) The archive contain an installer that need to be executed and let do his job
+3) The archive contain a complex hierarchy that need do to be extracted and add specific folder to PATH env var
+
+## Solution Design
+
+A new installation flow need to be created to execute different installation strategies regarding these three cases above.
+
+First thing first, the archive need to be extracted in a specific path (TEMP folder can be a good candidate).
+
+Regarding case 1, installation flow check if TEMP folder contain a single binary and repack it
+into an MSI package (including setting PATH env var). Then, MSI installation flow can take control.
+
+Regarding case 2, installation flow check if TEMP folder contain an installer (.exe, .msi, .msix) and pass control
+to MSI/Other installation flow.
+
+Regarding case 3, installation flow check if TEMP folder does not contain an installer nor a single binary.
+This case is the complex one, example with 'vagrant' hierarchy:
+- root
+  - bin 
+    - vagrant.exe
+    - other.exe
+  - embedded
+    - bin
+    - some other folders
+
+This hierarchy can be packed into an MSI installer, but how can we specify which binary to add 
+in PATH env var (in that case bin/vagrant.exe)?
+
+Take a look into the following manifest examples (v1.1.0)
+
+### Case 1|2 YAML manifest
```

Comment:
I would argue for case 2 to be the one that includes a relative path to the installer, in the event that there are support files included within the archive.

Reply:
Case 2 expect to have an installer at the root archive. But if the installer is located to 'archive.zip/somefolder/installer.exe' that should be a new case in my opinion. To avoid having more case I suggest to add a new attribute like BinaryPath.